pygrocy/grocy.py

```diff
@@ -99,9 +99,7 @@ def note(self) -> str:
         return self._note
         
     @property
-    def product(self) -> Product:
-        if self._product_id is None:
-            self.get_details()
+    def product(self) -> ProductData:
```

Comment:
I guess it hasn't been enforced in the past, but I initially planned to separate the server data models (stuff ending in Response or Data) from the public interface data models. With this server side changes to not always result in API changes, making it easier to maintain compabilitity. I have to think about this, maybe you could clarif this change.

Reply:
The code was wrong actually and didn't do nothing... I saw it during testing in coverage report (HTML report is nice :) )

About API change and data model, I understand what's your point.

Have a look to this API : https://github.com/kiwiz/gkeepapi
It uses node for data modeling and request modeling.